### ISSUE TITLE
Astropy Affiliated Package Review

### ISSUE THREAD

**Issue by astrofrog** [Chandra-MARX/marxs#167]
This package has been reviewed by the Astropy coordination committee for inclusion in the Astropy affiliated package ecosystem.

We have adopted a review process for affiliated package that includes assigning quantitative ‘scores’ (red/orange/green) for different categories of review. You can read up more about this process [here](https://docs.google.com/document/d/15PJf2PROXMa7xwTDvWnjXB_9KNuO2Ia4_kkxJ7MPazE/edit). (This document, currently in Google Docs, will be moved to the documentation in the near future.) For each of the categories below we have listed the score and have included some comments when the score is not green.

<table>
<tr><td><b>Functionality/Scope</b></td><td><a href="https://docs.google.com/document/d/e/2PACX-1vRXvsE3m8BBo6qE5pqUMfHAfHCDgRkOy4TumaHAaAt1jXMb_jHkNFFT1ddaTHnuzlP11dZxpQ_fe_BZ/pub#h.pch9wiwetm2l"><img src="https://img.shields.io/badge/Functionality-Specialized%20package-brightgreen.svg" alt="Specialized%20package"></a></td></tr>

<tr><td colspan=2><i>No further comments</i></td></tr>

<tr><td><b>Integration with Astropy ecosystem</b></td><td><a href="https://docs.google.com/document/d/e/2PACX-1vRXvsE3m8BBo6qE5pqUMfHAfHCDgRkOy4TumaHAaAt1jXMb_jHkNFFT1ddaTHnuzlP11dZxpQ_fe_BZ/pub#h.lizp0k629jgg"><img src="https://img.shields.io/badge/Astropy%20integration-Good-brightgreen.svg" alt="Good"></a></td></tr>

<tr><td colspan=2><i>No further comments</i></td></tr>

<tr><td><b>Documentation</b></td><td><a href="https://docs.google.com/document/d/e/2PACX-1vRXvsE3m8BBo6qE5pqUMfHAfHCDgRkOy4TumaHAaAt1jXMb_jHkNFFT1ddaTHnuzlP11dZxpQ_fe_BZ/pub#h.prawoqxdnwj5"><img src="https://img.shields.io/badge/Docs-Good-brightgreen.svg" alt="Good"></a></td></tr>

<tr><td colspan=2><i>No further comments</i></td></tr>

<tr><td><b>Testing</b></td><td><a href="https://docs.google.com/document/d/e/2PACX-1vRXvsE3m8BBo6qE5pqUMfHAfHCDgRkOy4TumaHAaAt1jXMb_jHkNFFT1ddaTHnuzlP11dZxpQ_fe_BZ/pub#h.8dots8fmtpvp"><img src="https://img.shields.io/badge/Tests-Good-brightgreen.svg" alt="Good"></a></td></tr>

<tr><td colspan=2><i>No further comments</i></td></tr>

<tr><td><b>Development status</b></td><td><a href="https://docs.google.com/document/d/e/2PACX-1vRXvsE3m8BBo6qE5pqUMfHAfHCDgRkOy4TumaHAaAt1jXMb_jHkNFFT1ddaTHnuzlP11dZxpQ_fe_BZ/pub#h.zh2jxv4intgn"><img src="https://img.shields.io/badge/Development-Good-brightgreen.svg" alt="Heavy%20development"></a></td></tr>

<tr><td colspan=2><i>No further comments</i></td></tr>

<tr><td><b>Python 3 compatibility</b></td><td><a href="https://docs.google.com/document/d/e/2PACX-1vRXvsE3m8BBo6qE5pqUMfHAfHCDgRkOy4TumaHAaAt1jXMb_jHkNFFT1ddaTHnuzlP11dZxpQ_fe_BZ/pub#h.9tur152fuipe"><img src="https://img.shields.io/badge/Python%203-Good-brightgreen.svg" alt="Good"></a></td></tr>

<tr><td colspan=2><i>No further comments</i></td></tr>

</table>

**Summary/Decision**: Things are looking good, and this package meets the review criteria for affiliated packages, so we are happy to confirm that we'll be listing your package as an affiliated package! Keep up the good work, and we encourage you to improve on the areas above that weren't “green” yet.

If you agree with the above review, please feel free to close this issue. If you have any follow-up questions or disagree with any of the comments above, leave a comment and we can discuss it here. At any point in future you can request a re-review of the package if you believe any of the scores should be updated - contact the coordination committee, and we’ll do a new review. Note that we are in the process of redesigning the http://affiliated.astropy.org page to show these scores (but not the comments). Finally, please keep the title of this issue as-is (“Astropy Affiliated Package Review”) to make it easy to search for affiliated package reviews in future.

**Comment by hamogu** [comment]
Comment: While not complete, there is considerable testing for correctness in the testing suite, e.g. in https://github.com/Chandra-MARX/marxs/blob/master/marxs/optics/tests/test_grating.py
where most of the tests compare results to the known good analytical solution of the grating equation or in https://marxs.readthedocs.io/en/latest/examples.html#polarization where simulation results are compared to experimental data.

I also want to remark that test coverage is higher when tests are run locally, because `visualization.mayavi` cannot even be imported on headless serves like Travis. (It's technically possible to set that up, but it's not easy.)

**Comment by astrofrog** [comment]
@hamogu - thanks for the clarification! We'll take a look at the testing aspect again. Regarding ``visualization.mayavi``, have you tried setting ``SETUP_XVFB=True`` on Travis? (should allow GUIs/VTK/OpenGL to run)

**Comment by astrofrog** [comment]
@hamogu - after discussing this with the coordination committee we've decided to bump up your 'test' score to green :) However I'd still encourage you to look into using ``SETUP_XVFB=True`` on Travis to try and run the mayavi-related tests.

**Comment by hamogu** [comment]
@astrofrog I did not know about the `SETUP_XVFB` option. I looked into installing xvfb myself, but decided that it's not worth it to require `sudo=True` just for this case.

That shows how much packages can profit from an independent review every now and then, particularly small packages that don't have the same range of experiences in the developer community that astropy itself has. (In this case, it's me with small contributions from two undergraduate students in their summer projects).

**Comment by hamogu** [comment]
@bsipocz : Right, this is exactly what I said "I did not know about the SETUP_XVFB option." **Instead** (and that is a word I should have added to my previous comment, but did not) I looked into installing xvfb myself, but decided that it's not worth it...".

I was under the impression (but that might be wrong, too), that I need to do `sudo = True`, if I want to `sudo apt-get install xvfb` and `set DISPLAY ...`. So, thank you to @astrofrog who pointed out that I can just try `SETUP_XVFB=True`.
Now, fortunately, that mean it does not matter any longer if I am right or wrong about sudo for apt-get install, because there is this other pathway to hopefully get what I want.
